UCP/WIREUP/SELECT: refactor ucp_wireup_select_ctx_t

evgeny-leksikov opened the pull request:
## What
Split ucp_wireup_select_ctx_t on ucp_wireup_select_context_t and
ucp_wireup_select_params_t

## Why ?
to separate in/out parameters in selection logic stack

@dmitrygx

swx-jenkins commented:
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/9251/ for details.

dmitrygx commented on src/ucp/wireup/select.c:
```diff
@@ -1377,20 +1423,26 @@ ucp_wireup_construct_lanes(ucp_wireup_select_ctx_t *select_ctx,
+        ucp_wireup_select_params_init(&select_params, ep, ep_params,
+                                      ep_init_flags, address_count,
+                                      address_list, worker->scalable_tl_bitmap);
+        memset(&select_ctx, 0, sizeof(select_ctx));
```
move `memset(&select_ctx, 0, sizeof(select_ctx));` to `ucp_wireup_search_lanes()`?

evgeny-leksikov replied:
ok, I was hesitate about this

dmitrygx replied:
> ok, I was hesitate about this

it's up to you anyway

dmitrygx commented on src/ucp/wireup/select.c:
```diff
@@ -74,27 +74,32 @@ typedef struct {
+    ucp_wireup_lane_desc_t    lane_descs[UCP_MAX_LANES]; /* Array of active lanes that are
+                                                          * found during selection */
+    ucp_lane_index_t          num_lanes;                 /* Number of active lanes */
+    unsigned                  ucp_ep_init_flags;         /* Endpoint init extra flags */
```
why need `ucp_` prefix?

evgeny-leksikov replied:
to avoid confusion near \*lane\* vars which are about UCT EPs

dmitrygx replied:
> to avoid confusion near *lane* vars which are about UCT EPs

ah, got it

dmitrygx commented on src/ucp/wireup/select.c:
```diff
@@ -673,8 +682,8 @@ ucp_wireup_add_memaccess_lanes(ucp_wireup_select_ctx_t *select_ctx,
 
-    while (select_ctx->address_count > 0) {
-        status = ucp_wireup_select_transport(select_ctx, &mem_criteria,
+    for (;;) {
```
👍 good catch
it was before changes from #4071 get merged

evgeny-leksikov commented:
@dmitrygx thanks for review! I've added missed but related refactoring in https://github.com/openucx/ucx/pull/4231/commits/f873332fafa92e77ba706de2f5a40f93285f92f6 and addressed your comment in https://github.com/openucx/ucx/pull/4231/commits/ef816465ae90bdbf587f12db7e80ba1c8d33dcae

pls take a look again

dmitrygx commented on src/ucp/wireup/select.c:
```diff
@@ -245,42 +288,38 @@ ucp_wireup_select_transport(const ucp_wireup_select_params_t *select_params,
-    show_error          = (select_params->show_error && show_error);
-    address_list        = select_params->address_list;
-    address_count       = select_params->address_count;
+    ucp_wireup_select_info_t sinfo;
```
`sinfo = {0};` and move up and remove `memsert()`

evgeny-leksikov replied:
See comment near `memset`, I tried to document that only initializing `sinfo.reachable = 0` is important, other fields could be uninitialized but compilers warn about this.

mellanox-github commented:
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12443/ for details (Mellanox internal link).

mellanox-github commented:
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 03-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/358989c0ae4cb06529b53dbd3e3c6f4b1002fe9e/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/4047f2a13085736fe3ca91365b7452f27dbfb015/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/1133edf084e7574a58a3f258ed693890696fcac9/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/2ee653d0dee09c6018ec112a5b69a0aef041c346/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/c64b9771c3a1d659de5e9367e386bbca1c95d09e/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/45e1fbcdbba27712c88b8faa50db4a441cdd91cf/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/714b7b54fd2765a0e494abc007a6396f17a01a01/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/bf30b98d282f714e17b83db7922fe93d09ed5269/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/7ca1527443d4ab63a54e6fbd40aa562ce10b7f2e/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/8a3e37b8dafbb7eb8ba1ba2823995ea3d7242339/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/eca1a69053c06ebc8403fea9ff5ede22dafe1e1f/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/e9c1f62511b8344fc019f2089c84b8aba6978bc9/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/d6d804f8ddeeefbe126da65c8e9f24d1fcf19d56/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/37b781e2d2ab21087da2ebce4645ba137f3841ae/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/71135770eae78643153685320f6274f5346104dc/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/915da5fcbb27785f6ee4cd1e899f36517f8f9d3d/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/4343325eb67b78dc781246508dd0c44822ae1d32/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/c62c13748a5331a4b70fadcc0ec3c2ce51cd12e1/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/4bff831201cc18ada4f8acc853dcf67f99725025/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/e28d2a934be8804a4b4ac8cc606f8fd5ed76a306/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/0d81007923606a0af94a754c67ac737c493e14f5/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/3673c33fb1f596bd7b2d7c9c682c24b078a975a6/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/33dc9f48973917262f355788a84f2cb677dac094/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/d96ea199c377664cd0e71eac2cf802a4f7503658/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/52f07b79dd0d75410ac44b76418287bd660e80d3/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/f9dd9f24cb14670bf324b4dc640c49547004b2aa/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/656d67ea3244c55caecc990b11b0d82a6f5a46c0/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/5dae6b48ea36724e86a0249cf12414a8e316cbd5/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9ee41996dde16473e4b14c48e000d40a/raw/eb10579ff3686655cef4a6bb034e7316d6d16fb5/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

dmitrygx commented on src/ucp/wireup/select.c:
```diff
@@ -233,6 +233,49 @@ static int ucp_wireup_check_amo_flags(const uct_tl_resource_desc_t *resource,
+              UCT_TL_RESOURCE_DESC_ARG(&context->tl_rscs[rsc_index].tl_rsc),
+              addr_index, criteria->title, score, priority);
+
+    if (update) {
```
it seems that `reachable` field is always 0

swx-jenkins commented:
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/9257/ for details.

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12449/ for details (Mellanox internal link).

mellanox-github commented:
<details><summary>Mellanox CI: <b>FAILED</b> on 6 of 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 03-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/196a03b33431a8a00883feae1a22b675067ac2ff/_main) | :x: FAILURE
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/3235c793f4cac08d447dda2465871252a4ea91fa/hpc-test-node-legacy_W0) | :x: FAILURE
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/573a1547d504e34b5361fe755d1c256ca6796df7/hpc-test-node-legacy_W3) | :x: FAILURE
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/4763f8e1502760891d5fbf90dc929cda64644781/r-vmb-ppc-jenkins_W1) | :x: FAILURE
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/5afe3c481e1cf3ec39b3a45011a6a8396159670d/r-vmb-ppc-jenkins_W2) | :x: FAILURE
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/25a3b09fd8f77db255a623d743f85460e7f972f3/r-vmb-ppc-jenkins_W3) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/c02775c7e25b208f59f2246852b452f47bf2b263/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/a58dced81d3f7eee35d3f720060e0085b7a83ba3/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/7ec0e4891f4499f832991c4a177b297b8d544a9c/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/a4a2b046c9f54b464c8f82b39293231ad21f07d5/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/71c5bae587a7feec4b0d86c2a96cd3a9d2fcf97f/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/029a31c797eddaea59383daf6397eab00ac11b62/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/df5d91da907d1c21b4d99902d53d60e13d023cd3/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/4aa104bdd816b9cd2650e70ce398d103369ce54d/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/68feeca53462a4904253a3c4351bf7c47c0a887c/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/a6d2a6347920d6ad15cd3056ffc35d779ec7baef/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/f225d008f4d793146c92c67289f037f49aad0e2f/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/f0b6a33526377ec08c9613d60db27937a9d9b2ad/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/0939c57a45c5e317380cf16644d0afc4b6d4da93/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/33f2686d4f105abb5714156efb60f0eec969ba23/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/30d2b88052ea03a44ca798f33a935de548eb60c0/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/53e0836c543291737cddc113c00d8d4081d41cf9/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/90312cfc4af4b05d64a52c8ecc85589999a53cee/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/f0de6578b2319dcf0969406dc485ddf974db623f/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/ab024d50032f5d940857909469154e4fcbbcde4d/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/2c5ce78170b9d7df7dc7703a29f279d3d106943d/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/a83290ea0b9917959404262e221a32e9aed0209e/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/979aaaf3cfbdaf0a5ac7ed9f9a69bed6d8bbdfc9/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b6e412b209c4a106e56352f5bccbcf6a/raw/d18d03b8d099dcdaaad046fb406e5897541713e4/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS

swx-jenkins commented:
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/9258/ for details.

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12450/ for details (Mellanox internal link).

mellanox-github commented:
<details><summary>Mellanox CI: <b>FAILED</b> on 2 of 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 03-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/22b3b832e746dd116050b6d1ff5ae783a519e778/_main) | :x: FAILURE
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/41f314694e1bc4ad90230f81539360fc15a01139/hpc-test-node-legacy_W0) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/ca722ff1f925f49f2e82e95f3a138c2ff95a1150/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/d0443b2bf5cc180561b08b89b9711ea3bdeb3b26/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/aa7f8af911b9ec44aa9c46d6992350db593fed08/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/9d176a08cd6d810c947e9b00d7b521a420c10ab3/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/0412a26deba8324d81520826b1895fc19f6f5694/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/59cd6bf77e61069be96d10e778c2165baaf53797/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/0eb3b4e14ecb7bbdc12c41055d008124ddc86ed2/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/c89f6182e40034363cb446ffbdc78cdf41887451/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/872fbb0e54d65383937f39a2332cd27e43b664af/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/53e002823c8a2634ae506a6ad934e7bab4433c34/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/a468d57260d68eaea50fe878bbbad4fd03c1ca3b/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/e05517d2c09cc8b42d9206a7904b9d24768c1eeb/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/60e3c320193bb6d46715b5f6b719e0426618090e/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/f520fe71fd6944278fdffcaf6c78e9a0f2204d63/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/c0576a5b2968f86fe2acedf758034e655347d3c3/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/bc92b7a17340372e747fd8d4a2089bcdad4876cd/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/4832d3c3570900f3f943328a17707ea5649542c4/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/62fc757b7859533a5c50953daf2df8e9b201a6af/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/5c40c6263f4b57137947e5801e90e718eca35c75/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/f000476912a4bdd3f2257f57a7046e22df5535c2/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/41f3d913fd06cbf3dd6d542f0bcd1064c3a81439/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/4b7ea5d3546655933891a51cf4d0d570faa79614/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/453cf99dc3fdabbd0ed8ef44c0ff17bed11373a6/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/2f97cb8a09a47900b8169e09e3d2d46861fe9e45/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/f5563538199c92f96f3b600a5df7661d1faec91f/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/87bb6f55f4292e5be14ae7a8b10720c0e442e4a4/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/2819f11eff29f279633e8c09c58b590f/raw/2932ecb9b0a882febe40dd40ae63c9079d370125/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

evgeny-leksikov commented:
failures are https://github.com/openucx/ucx/issues/4207
bot:mlx:retest
bot:pipe:retest

mellanox-github commented:
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12455/ for details (Mellanox internal link).

mellanox-github commented:
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 03-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/cdf1a2eefeadc86424f8647bccb71a44a5639eb7/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/0be362cd6e0bc450c972de6aac7bf74996d30616/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/13c005cfef47a66bbb1a139c2c5cf46abe04a84c/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/2f264e08d95d342e78ead8fb44e3770b01e26d6f/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/457e32105ab141ad450e6b7cf66d5ea3d9e71144/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/71d992cb31be062b3c486c18fd3099b6201478e4/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/4be63e6a996d26bc16ffccfd8b23c8880ffd74f2/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/0c9ffcf674908a965394c92a77a24ebaaa9a67dc/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/e7634eaf7b4b165bc5066a6eee713a7bff274091/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/c7c7ba9a92f8bfc82b869e1ece9dcedbca89c8f5/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/63fb403bd205ae65acd9d4d85ce92bc7dc821f63/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/8792e0653ade3a6cee6a68ea393ae04b9e1de2ca/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/bf3ee7d8f33a5ff486d380f76f7ada5ff6cbc100/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/dd6f81884f4916a6353594f3246f69ba1d50e00f/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/9fc03f071fd09906e585781f397a932daf185c33/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/518c9534806bd50aff8574f7df1e17d30f03926e/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/f6833fea676c32b1d587bd68cc7022b9af06642b/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/111857ebcbfeb8a458bfb618684ba47089ed9cbd/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/3e0a9e907fb115eef2d4aca1bdc68a96b263b0d2/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/5b697cdc66cd9b6c3f931c5fa8fb2a22b3629a28/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/08b5d183397657a911120b3d7977907164701b50/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/64d2e8096a8e6855f458befda98acda949c5c35d/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/bff58a769e5d18cb19f3f528679339b4c00851cd/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/22dd3ef6737ce887c3eb5e2056cce3e2f605dd42/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/300c0143f28b2e4907698114e71d886b000a5374/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/05bb6b323cae36615005b58b21ba7b817b1eb688/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/606762f5b0a58141a2e61cc3d38e57b983c7ab67/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/584b59268bebd38898377fa232c6d8b1cf459cba/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/5bd4a54d5f13add77268298128a557fa/raw/b369c1190f6d6f6566ac6336023732d1e0ca8d84/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

brminich commented on src/ucp/wireup/select.c:
```diff
@@ -74,27 +74,32 @@ typedef struct {
                                                           * procedure */
-    ucp_wireup_select_info_t  am_info;                   /* AM transport selection info */
-} ucp_wireup_select_ctx_t;
+} ucp_wireup_select_params_t;
```
maybe keep am_info here?

evgeny-leksikov replied:
This structure is initialized once, am_info is a result of AM selection logic which goes further only to TM selection logic.

brminich commented on src/ucp/wireup/select.c:
```diff
@@ -1120,19 +1176,22 @@ ucp_wireup_add_rma_bw_lanes(ucp_wireup_select_ctx_t *select_ctx)
 static ucs_status_t
-ucp_wireup_add_tag_lane(ucp_wireup_select_ctx_t *select_ctx,
-                        ucp_err_handling_mode_t err_mode)
+ucp_wireup_add_tag_lane(const ucp_wireup_select_params_t *select_params,
```
in this func we have `select_params`, `select_info` and `select_context`, which is a bit confusing. 
better keep `am_info` inside `params`

evgeny-leksikov replied:
I would like to keep `params` as a const input parameters everywhere, `ucp_wireup_select_info_t am_info`, which is a result of AM selection logic, is needed only for TM fallback. maybe renaming `am_info` to `fallback_am_info` would make it a bit clearer, WDYT?

alinask replied:
@brminich can you please reply?

evgeny-leksikov commented:
@alinask `mpirun -mca mpi_add_procs_cutoff 100000 -mca pml ucx IMB-MPI1 -npmin 224` on 8 hosts with full ppn works as expected.
